src/OpenTelemetry/Metrics/Exemplar/SimpleExemplarReservoir.cs

```diff
@@ -57,8 +57,12 @@ public override Exemplar[] Collect(ReadOnlyTagCollection actualTags, bool reset)
             if (reset)
             {
                 this.runningExemplars[i].Timestamp = default;
-                this.measurementsSeen = 0;
             }
+
+            // Reset internal state irrespective of temporality.
+            // This ensures incoming measurements have fair chance
+            // of making it to the reservoir.
+            this.measurementsSeen = 0;
```

Comment:
Should this be reset on every iteration of the for loop?

Reply:
good catch! fixed.